digits/frameworks/exceptions.py

```diff
@@ -0,0 +1,10 @@
+from digits.utils import subclass, override
+
+@subclass
+class BadNetworkException(Exception):
```

Comment:
Having custom exceptions for the Frameworks module is awesome!

Can we follow [this convention](https://docs.python.org/2/tutorial/errors.html#user-defined-exceptions) and create two classes with these names?
```python
class Error(Exception):
    pass
class BadNetworkError(Error):
    pass
```
I know we haven't been consistent with this before, but I'd like to move in that direction. See here:
https://github.com/NVIDIA/DIGITS/blob/4121e3a1c0/tools/create_db.py#L41-L58